apps/researcher/src/lib/community.ts

```diff
@@ -1,24 +1,79 @@
 import {clerkClient, auth} from '@clerk/nextjs';
 import {OrganizationMembership, Organization} from '@clerk/backend/dist/types';
 
-export type Community = Organization;
+export interface Community {
+  id: string;
+  name: string;
+  description?: string;
+  slug: string;
+  imageUrl: string;
+  createdAt: number;
+}
+
+export interface Membership {
+  id: string;
+  role: string;
+  userId: string;
+  firstName: string;
+  lastName: string;
+  imageUrl: string;
+}
+
+function organizationToCommunity(organization: Organization): Community {
+  return {
+    id: organization.id,
+    name: organization.name,
+    // The type of `publicMetadata` is `{ [k: string]: unknown } | null `. Redeclare custom metadata `description`.
+    description: organization.publicMetadata?.description as string | undefined,
+    slug: organization.slug!,
+    imageUrl: organization.imageUrl,
+    createdAt: organization.createdAt,
+  };
+}
+
+function organizationMembershipToCommunityMembership(
+  membership: OrganizationMembership
+): Membership {
+  if (!membership.publicUserData) {
+    throw new Error('publicUserData is undefined');
```

Comment:
Minor: this throws when `publicUserData` is undefined. But if it is defined, lines 44 to 48 assume that it is of a certain type, e.g. it has props such as `userId`. Is this always the case? If so, make this clear in a comment? If this is not the case, validate the `publicUserData` with zod?

Reply:
I have rewritten this function a bit. I have set all my assumptions about the Clerk data in a code comment and let typescript throw errors by adding '!' to the code.